_posts/2022-12-08-developer-survey-2022.md

```diff
@@ -0,0 +1,123 @@
+
+## DevOps
+
+For version control, git has swept the floor - in the survey we did not have a single vote for any other version control system. In the VCS service selection, there was more variation. GitHub was the most used (Solita is a GH corporate customer). Where the customer has other preferences, we have used Azure DevOps Services, GitLab and BitBucket.
+
+In the devops section we had an open question about the use of git hooks in the daily work.
+
+The build and deployment pipelines are most often built with either Azure Pipelines (35%), Jenkins (28%), or GitHub Actions (20%). Some projects use GitLab CI (7%), or do deployments with custom scripting (5%). The rest consists of 4-5 votes per platform in the list of CircleCI, BitRise, Octopus Deploy and TeamCity.
+
+For infrastructure management, the top votes spread over various shell/makefile/yaml/powershell solutions, Terraform, Ansible and AWS CDK. Some projects make use of other Azure products like CloudFormation and Resource manager or Google Cloud Deployment Manager.
+
+Majority of the projects have automated deployment of the software to test environments. For production environments, it is more common to have automated deployment with a manual trigger, but 25% of respondents indicate that in their projects also the production deployment is fully automatic.
+
+## Testing & Monitoring
+
+While in the title catalog in the Solita developer community we do not have many people dedicated testing roles, we do have proper focus in the maturity and quality of the code that we produce. Our teams contribute in all testing activities from smoke testing to regression/acceptance tests and we also conduct penetration testing and A/B testing for some of our customers.
+
+In the past two years the most used test frameworks are Cypress (31%), JUnit (30%) and Jest (21%). Besides unit testing, UI testing and E2E testing are done with e.g. Robot Framework, Puppeteer and Cucumber frameworks.
+
+Automation is heavily used in the testing. 79% have automated unit tests, 63% integration tests, 48% E2E tests. Around 10% of the respondents indicate that in their projects the automation extends to performance and security testing.
+
+## Trends & Insights
+
+During the past two years, Solita Developer community has grown significantly both through organic new hires and acquisitions. Particularly Commentor AS from DK joining Solita brought in a merry lot of MicroSoft/Azure stack users (and about 70 respondents in the survey). This has an impact on multiple levels, and it shows also in this survey in the response spread for many questions, starting from the most common workstation OS, which tipped towards Windows. Solita's Mobile development has been traditionally Finland-originating endeavour, but with Commentor, we now also have several people working with mobile stacks in Denmark.
```

Comment:
Few suggestions here:

"from DK" => "from Denmark" to make it easier to read
MicroSoft => Microsoft

Reply:
Fixed